### PR TITLE
feat: add implementation of `stats/strided/diqrsorted`

### PR DESCRIPTION
**Pull request by Om-A-osc**
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: passed
  - task: lint_c_examples status: passed
  - task: lint_c_benchmarks status: passed
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed
   ---
Resolves #NA

## Description

> What is the purpose of this pull request?

This pull request:

-   adds an implementation of `@stdlib/stats/strided/diqrsorted`
-   computes the interquartile range (IQR) for sorted double-precision
    floating-point strided arrays
-   leverages the existing `dmediansorted` implementation to compute
    the first and third quartiles, ensuring numerical correctness and
    consistency with existing sorted statistics utilities
-   provides both main and `.ndarray` interfaces following stdlib
    strided API conventions
-   includes comprehensive unit tests covering edge cases, stride,
    negative stride, and offset semantics
-   adds benchmarks for performance evaluation
-   includes a C implementation and corresponding examples to maintain
    parity with other strided statistical routines
-   adds REPL help, documentation, and examples
-   follows stdlib formatting, linting, and code style guidelines

This addition extends the sorted statistics utilities by providing a
robust and efficient measure of statistical dispersion.

## Related Issues

> Does this pull request have any related issues?

This pull request has the following related issues:

-   NA

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Implementation notes:

-   The implementation assumes the input array is already sorted.
-   Quartiles are computed using the median-of-halves approach,
    consistent with common statistical definitions and boxplot usage.
-   Reusing `dmediansorted` reduces duplication and ensures
    consistency with existing sorted median behavior.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

> When authoring the changes proposed in this PR, did you use any kind of AI assistance?

-   [ ] Yes
-   [x] No

#### Disclosure

N/A

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/strided/diqrsorted/docs/repl.txt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    Examples
+    --------
+    // Standard usage:
+    > var x = new {{alias:@stdlib/array/float64}}([ 1, 2, 3, 4, 5 ]);
+    > {{alias}}( x.length, x, 1 )
+    3.0
+
+    // Using `N` and stride parameters:
+    > var x = new {{alias:@stdlib/array/float64}}([ 1,0, 2,0, 3,0, 4 ]);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/strided/diqrsorted/docs/repl.txt]: Looks like commas were used instead of decimal points here -- `1,0` parses as two separate values (`1` and `0`) rather than the float `1.0`. It works by coincidence because the stride pattern still picks the right sorted values, but this should use proper float notation:

```suggestion
    > var x = new {{alias:@stdlib/array/float64}}([ 1.0, 0.0, 2.0, 0.0, 3.0, 0.0, 4.0 ]);
```

File: lib/node_modules/@stdlib/stats/strided/diqrsorted/docs/repl.txt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    > {{alias}}( 4, x, 2 )
+    2.0
+
+    // Using view offsets:
+    > var x0 = new {{alias:@stdlib/array/float64}}([ 0,1, 2,3, 4,5 ]);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/strided/diqrsorted/docs/repl.txt]: Same comma-instead-of-period issue here. `0,1` parses as two values instead of the float `0.1`. This should use proper float notation:

```suggestion
    > var x0 = new {{alias:@stdlib/array/float64}}([ 0.0, 1.0, 2.0, 3.0, 4.0, 5.0 ]);
```
